crates/chia-sdk-driver/src/primitives/action_layer/precommit_coin.rs

```diff
@@ -0,0 +1,165 @@
+use chia_protocol::{Bytes32, Coin};
+use chia_puzzle_types::{
+    cat::{CatArgs, CatSolution},
+    CoinProof, LineageProof,
+};
+use chia_sdk_types::puzzles::PrecommitLayerSolution;
+use clvm_traits::ToClvm;
+use clvm_utils::TreeHash;
+use clvmr::{Allocator, NodePtr};
+
+use crate::{CatLayer, DriverError, Layer, PrecommitLayer, Spend, SpendContext};
+
+#[derive(Debug, Clone)]
+#[must_use]
+pub struct PrecommitCoin<V> {
+    pub coin: Coin,
+    pub asset_id: Bytes32,
+    pub proof: LineageProof,
+    pub inner_puzzle_hash: Bytes32,
+
+    pub controller_singleton_struct_hash: Bytes32,
+    pub relative_block_height: u32,
+    pub payout_puzzle_hash: Bytes32,
+    pub refund_puzzle_hash: Bytes32,
+    pub value: V,
+}
+
+impl<V> PrecommitCoin<V> {
+    #[allow(clippy::too_many_arguments)]
```

Comment:
It'd be nice to use struct(s) to group arguments instead of disabling this lint

Reply:
no